client.go

```diff
@@ -115,6 +124,29 @@ func (c *Client) createToken(params map[string]interface{}, expire time.Time) ([
 	return claims.HMACSign(jwt.HS256, c.apiSecret)
 }
 
+// WithTimeout sets http requests timeout to the client
+func WithTimeout(t time.Duration) func(*Client) {
```

Comment:
what do you think about this one? 
https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
Options are a necessary evil. Do we really need options? I'd expose the flexible http.Client only.

Reply:
I think it's pretty good way for customizable constructors. It's better than exposed fields, because it can do some stuff like setters does, and also looks cleaner than setters